### PR TITLE
Add Workflow to Flathub

### PR DESCRIPTION
**Pull request by cunidev**
First attempt at flatpak-ing an app, so apologies if there are any mistakes in the procedure.

### CODE REVIEW

File: com.gitlab.cunidev.Workflow.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+    "app-id" : "com.gitlab.cunidev.Workflow",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.32",
```
Comment (bilelmoussaoui) [inline com.gitlab.cunidev.Workflow.json]: Have you tried using 3.34?

File: com.gitlab.cunidev.Workflow.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    "tag": "0.1.0",
+                    "commit": "580d73ffe199fb784989fa3f640da2d12da35eab"
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {
+        }
+    }
```
Comment (bilelmoussaoui) [inline com.gitlab.cunidev.Workflow.json]: You can remove these.

File: com.gitlab.cunidev.Workflow.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "/include",
+        "/lib/pkgconfig",
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
```
Comment (bilelmoussaoui) [inline com.gitlab.cunidev.Workflow.json]: You probably don't need any of those cleanups.

File: com.gitlab.cunidev.Workflow.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "modules" : [
+        {
+            "name" : "Workflow",
+            "builddir" : true,
```
Comment (bilelmoussaoui) [inline com.gitlab.cunidev.Workflow.json]:
```suggestion 
```

File: com.gitlab.cunidev.Workflow.json (hunk 1 above)
Comment (bilelmoussaoui) [reply com.gitlab.cunidev.Workflow.json]: Just restart Builder and it should work.

Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: Thanks. I tested it and it seems to work just as expected on 3.34

File: com.gitlab.cunidev.Workflow.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "workflow",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/activitywatch",
+        "--filesystem=~/.config/dconf:ro",
```
[the remark this answers is not in the document]
Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: So neither line 12 nor 14?

[the remark this answers is not in the document]
Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: I've removed both `~/.config/dconf:ro` and `=xdg-run/dconf`, please let me know if that is not correct

**Comment by LeandroStanger** [comment]
bot, build com.gitlab.cunidev.Workflow

**Comment by cunidev** [comment]
Oh. I'll revise the AppData and fix this within days.

**Comment by cunidev** [comment]
Should be fixed in theory.

**Comment by bilelmoussaoui** [comment]
bot, build com.gitlab.cunidev.Workflow

File: com.gitlab.cunidev.Workflow.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "workflow",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=~/.config/activitywatch",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (bilelmoussaoui) [inline com.gitlab.cunidev.Workflow.json]: Those are part of dconf too. They can be removed.

**Comment by cunidev** [comment]
@nedrichards is awaiting-changes outdated at this point?

**Comment by bilelmoussaoui** [comment]
bot, build com.gitlab.cunidev.Workflow

**Comment by cunidev** [comment]
In the meanwhile I've updated this to 0.1.1 - should correct an annoying glitch with the calendar widget

**Comment by LeandroStanger** [comment]
bot, build com.gitlab.cunidev.Workflow

File: com.gitlab.cunidev.Workflow.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.34",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "workflow",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=~/.config/activitywatch"
```
Comment (barthalion) [inline com.gitlab.cunidev.Workflow.json]: Do you actually need this? Your application should respect $XDG_CONFIG_DIR environment variable which is overwritten by flatpak to point to correct, sandboxed directory.

Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: And that folder belongs to an external tool Workflow cannot do without.

Comment (bilelmoussaoui) [reply com.gitlab.cunidev.Workflow.json]: Does the application display anything in case activitywatch is not installed?

Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: Yes, a generic "empty state" recommending to set it up properly if it's not installed/not running so far. Better error handling planned

Comment (cunidev) [reply com.gitlab.cunidev.Workflow.json]: @barthalion

Comment (barthalion) [reply com.gitlab.cunidev.Workflow.json]: I guess that's fine then.

**Comment by barthalion** [comment]
I see appdata also mentions that AW needs to be installed so we're good.

/merge

**Pull request closed without merging**